### PR TITLE
Disable binary caches for CI

### PR DESCRIPTION
**Pull request by mrks**
For the join order benchmark, binary tables are written into the table_path. In CI, this path is shared by different targets that are potentially executed concurrently. This sometimes led to issues with corrupted binary files.

I am also sneaking in some things that are too minor to have their own PR.

### CODE REVIEW

File: scripts/test/hyriseBenchmarkJoinOrder_test.py [hunk 1]
```diff
@@ -29,6 +28,10 @@ def main():
     arguments["--clients"] = "1"
     arguments["--scheduler"] = "false"
 
+    # Binary tables would be written into the table_path. In CI, this path is shared by different targets that are
+    # potentially executed concurrently. This sometimes led to issues with corrupted binary files.
```
Comment (Bensk1) [inline scripts/test/hyriseBenchmarkJoinOrder_test.py:32]: This problem should occur for all benchmarks that cache binary tables, right? The other benchmark script files don't contain such a comment, so sad.

File: src/lib/operators/join_hash/join_hash_steps.hpp [hunk 2]
```diff
@@ -211,11 +211,12 @@ class PosHashTable {
 
 // The bloom filter (with k=1) is used during the materialization and build phases. It contains `true` for each
 // `hash_function(value) % BLOOM_FILTER_SIZE`. Much of how the bloom filter is used in the hash join could be improved:
-// (1) Dynamically check whether bloom filters are worth the memory and computational costs This could be based on the
+// (1) Dynamically check whether bloom filters are worth the memory and computational costs. This could be based on the
 //     input table sizes, the expected cardinalities, the hardware characteristics, or other factors.
 // (2) Choosing an appropriate filter size. 2^20 was experimentally found to be good for TPC-H SF 10, but is certainly
 //     not optimal in every situation.
-// (3) Check whether using multiple hash functions (k>1) brings any improvement.
+// (3) Check whether using multiple hash functions (k>1) brings any improvement. For single-threaded TPC-H SF 10 (which
```
Comment (Bensk1) [inline src/lib/operators/join_hash/join_hash_steps.hpp:218]: `where many are under 2^20`: does this mean that most of these integers have values < 1M?

Comment (Bensk1) [reply src/lib/operators/join_hash/join_hash_steps.hpp:218]: Also, `it however appears as if this has no immediate benefits.`: multiple hash functions have no effect or the check has no effect?

File: src/test/lib/concurrency/stress_test.cpp [hunk 3]
```diff
@@ -54,6 +54,8 @@ TEST_F(StressTest, TestTransactionConflicts) {
   };
 
   // Create the async objects and spawn them asynchronously (i.e., as their own threads)
+  // Note that async has a bunch of issues (cf. Mastering the C++17 STL, pages 205f).
```
Comment (Bensk1) [inline src/test/lib/concurrency/stress_test.cpp:57]: Hm, I don't really know what to make of that and if the reader of this comment will order the book to understand the comment. Is there an online version of it?

File: scripts/test/hyriseBenchmarkJoinOrder_test.py (hunk 1 above)
Comment (mrks) [reply scripts/test/hyriseBenchmarkJoinOrder_test.py:32]: Not an issue for the TPC-* benchmarks, which place the cached files in the current directory. You are right about the file-based benchmark though.

File: src/lib/operators/join_hash/join_hash_steps.hpp (hunk 2 above)
Comment (mrks) [reply src/lib/operators/join_hash/join_hash_steps.hpp:218]: > Evaluate whether using multiple hash functions (k>1) brings any improvement. In a first experiment, however, we saw no benefits for single-threaded TPC-H SF 10.

Better? I don't think we need to be too specific here.

File: src/test/lib/concurrency/stress_test.cpp (hunk 3 above)
Comment (mrks) [reply src/test/lib/concurrency/stress_test.cpp:57]: > Note that async has a bunch of issues:
> - https://stackoverflow.com/questions/12508653/what-is-the-issue-with-stdasync
> - Mastering the C++17 STL, pages 205f

better?

**Review by Bensk1 — APPROVED** [review]

**Pull request merged**